flutter_web_auth_2/android/src/main/kotlin/com/linusu/flutter_web_auth_2/CallbackActivity.kt

```diff
@@ -1,20 +1,54 @@
 package com.linusu.flutter_web_auth_2
 
+import android.annotation.SuppressLint
 import android.app.Activity
+import android.content.Intent
 import android.net.Uri
+import android.os.Build
 import android.os.Bundle
 
-class CallbackActivity: Activity() {
-  override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
+class CallbackActivity : Activity() {
 
-    val url = intent?.data
-    val scheme = url?.scheme
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
 
-    if (scheme != null) {
-      FlutterWebAuth2Plugin.callbacks.remove(scheme)?.success(url.toString())
+        val url = intent?.data ?: fixAutoVerifyNotWorks(intent)
+        val scheme = url?.scheme
+
+        if (scheme != null) {
+            FlutterWebAuth2Plugin.callbacks.remove(scheme)?.success(url.toString())
+        }
+        @SuppressLint("ObsoleteSdkInt")
+        if (Build.VERSION.SDK_INT >= 21) {
+            finishAndRemoveTask()
+        } else {
+            finish()
+        }
```

Comment:
```suggestion
        finishAndRemoveTask()
```

You should be able to assume that `SDK >= 21` as it is now [the minimum requirement](https://github.com/ThexXTURBOXx/flutter_web_auth_2/blob/master/flutter_web_auth_2/android/build.gradle#L48). The same holds true for [Flutter 3.22.0 itself](https://github.com/flutter/flutter/pull/142267) btw

Reply:
yes ,  finishAndRemoveTask() is ok